vscode/microsoft-kiota/src/util.ts

```diff
@@ -92,4 +95,127 @@ export async function handleMigration(
           vscode.window.showErrorMessage(vscode.l10n.t(`Migration failed: ${error}`));
       }
   });
-}
\ No newline at end of file
+}
+
+export function getSanitizedString(rawValue?: string): string| undefined{
+  return rawValue?.replace(/[^a-zA-Z0-9_]+/g, '');
+};
+
+export function parseGenerationType(generationType: string | QuickPickItem | undefined): GenerationType {
+    if(typeof generationType !== 'string') {
+        throw new Error('generationType has not been selected yet');
+    }
+    switch(generationType) {
+        case "client":
+            return GenerationType.Client;
+        case "plugin":
+            return GenerationType.Plugin;
+        case "apimanifest":
+            return GenerationType.ApiManifest;
+        default:
+            throw new Error(`Unknown generation type ${generationType}`);
+    }
+}
+
+export function parseGenerationLanguage(value: string): KiotaGenerationLanguage {
+    switch (value) {
```

Comment:
could it be easier to call something like `value.ToLowerCase()` and reduce the number of cases here.

Reply:
The existing cases are already applied in other places. Trying to reduce the refactor work on this PR. We'll see how much refactor will be included with this change.